rascaline/src/calculators/soap/power_spectrum.rs

```diff
@@ -677,73 +681,90 @@ impl CalculatorBase for SoapPowerSpectrum {
                     });
             }
 
-            // gradients with respect to the cell parameters
-            if let Some(mut gradient) = block.gradient_mut("cell") {
-                let gradient = gradient.data_mut();
+            // gradients with respect to the strain/cell
+            for parameter in ["cell", "strain"] {
```

Comment:
Shouldn't the list `["cell", "strain"]` be created from what the user is requesting? If he only want `"strain"` we should not do cell gradients

Reply:
This will be checked by the `if let Some(mut gradient) = block.gradient_mut(parameter)`. The block contains the gradients the user requested; and this function returns `None` if no gradient is defined.

Reply:
Okay makes sense.